cli/azd/cmd/sets.go

```diff
@@ -82,6 +88,15 @@ func newAzdContext() (*azdcontext.AzdContext, error) {
 	return azdCtx, nil
 }
 
+func newCredential() (azcore.TokenCredential, error) {
+	credential, err := azidentity.NewAzureCLICredential(nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating credential: %w", err)
```

Comment:
Maybe we can opt-for a more user-friendly error message here: "failed to obtain Azure credentials: %w"

Reply:
Done.